### DISCUSSION TITLE
is there a global hook to run when thrown HTTPError ?

### DISCUSSION THREAD

**Discussion by ma101an** [sindresorhus/ky#411] [Q&A] [answered]
when HTTPError happens, is there a global hook to execute instead of catching on every request?

use case:
- To send error details to error-tracker like sentry on any request failures

**Comment by sindresorhus** [comment, marked as answer]
Global hooks is an anti-pattern. Ky has local hooks where you can add a hook and then use `ky.extend()` to create a reusable instance with such hook, which you would put in a separate file an import in various places in your app instead of the `ky` import. Unfortunately, Ky does [not yet have a `beforeError` hook](https://github.com/sindresorhus/ky/issues/412). For now, you could just wrap Ky in an app-specific "request" method which you use throughout your app.

**Reply by ma101an** [reply]
Thanks. By global hook I actually meant the interface to execute some logic in such event as error. (ie. local hooks)
Right now I'm checking it in afterResponse hook (where I execute the logic inside `!response.ok` condition as a alternative approach)

**Reply by sholladay** [reply]
FYI to all, support for the `beforeError` hook has been released. In either a `beforeError` hook or an `afterResponse` hook, you can check if `error instanceof ky.HTTPError` to see if an error is specifically caused by a status code. I would advise against checking `response.ok`, since we do that internally for you when determining whether to throw an `HTTPError`.
